File: trunk/src/proof-of-concept/.NET (Windows Forms)/XMLBasedGUI/XmlBasedWindow.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Drawing;
using System.Windows.Forms;

namespace XMLBasedGUI
{
    class ControlFactory
    {
        public static Button createButton(String ID, Point topLeft, Size size, String label)
        {
            Button button = new System.Windows.Forms.Button();
            button.Location = topLeft;
            button.Name = ID;
            button.Size = size;
            //button.TabIndex = 0;
            button.Text = label;
            button.UseVisualStyleBackColor = true;
            return button;
        }

        public static TextBox createEditbox(String ID, Point topLeft, Size size, String text)
        {
            TextBox editbox = new System.Windows.Forms.TextBox();
            editbox.Location = topLeft;
            editbox.Name = ID;
            editbox.Size = size;
            //editbox.TabIndex = 0;
            editbox.Text = text;
            return editbox;
        }
    }

    public class XmlBasedWindow : Form
    {
        protected XmlBasedWindow(String pathToXML)
        {
            // we're working on the window, so lets not make the changes visible until we're completely done
            this.SuspendLayout();

            // read the XML and create the controls specified there
            parseXML(pathToXML);

            // okay, now lets make this visible
            this.ResumeLayout();
        }

        private void parseXML(String pathToXML)
        {
            // DEBUG: demo code (all the data here would normally be retrieved from the XML)
            //this.AutoScaleDimensions = new System.Drawing.SizeF(6F, 13F);
            //this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
            //this.ClientSize = new System.Drawing.Size(292, 273);
            this.Name = "test1";
            this.Text = "test1";
            this.Controls.Add(ControlFactory.createEditbox("text", new Point(10, 10), new Size(100, 25), ""));
            this.Controls.Add(ControlFactory.createButton("button", new Point(10, 50), new Size(100, 25), "Hello, world!"));
        }

        protected Control getControlByID(String ID)
        {
            foreach (Control c in this.Controls)
            {
                if (c.Name == ID)
                    return c;
            }
            return null;
        }

        abstract void onButtonClick(...);
    }
}

File: trunk/src/proof-of-concept/Python (wxWindows)/XMLBasedGUI/XmlBasedWindow.py
import wx

#from frame import Frame



class ControlFactory:
    @staticmethod
    def createButton(parent, ID, topLeft, size, label):
        button = wx.Button(parent, id=-1, label=label, pos=topLeft, size=size)
        button.name = ID
        return(button)
    
    @staticmethod
    def createEditbox(parent, ID, topLeft, size, text):
        editbox = wx.TextCtrl(parent, id=-1, pos=topLeft, size=size)
        editbox.SetValue(text)
        editbox.name = ID
        return(editbox)
        


class XmlBasedWindow(wx.Frame):
    def __init__(self, pathToXML):
        # create the frame
        wx.Frame.__init__(self, None, -1, "This is a test")
        #wx.PyWindow.__init__(self, None, -1, wx.DefaultPosition, wx.DefaultSize, wx.DEFAULT_FRAME_STYLE, "This is a test")
        #wx.Frame.__init__(self, None, "window", "title", wx.DefaultPosition, wx.DefaultSize, wx.DEFAULT_FRAME_STYLE)
        
        # create a panel inside the frame
        self.panel = wx.Panel(self, -1)
        
        # prepare private attributes
        self.controls = []
        
        # read the XML and create the controls specified there
        self.parseXML(pathToXML)

    def OnCloseWindow(self, event):
        self.Destroy()
        
    def parseXML(self, pathToXML):
        # create the controls
        editbox = ControlFactory.createEditbox(self.panel, "text", (10, 10), (100, 25), "")
        button = ControlFactory.createButton(self.panel, "button", (10, 50), (100, 25), "Hello, world!")
        self.controls.append(editbox)
        self.controls.append(button)
        #button = wx.Button(panel, 1003, "Close Me")
        #button.SetPosition((15, 15))
        
        self.Bind(wx.EVT_CLOSE, self.OnCloseWindow)

    def getControlByID(self, ID):
        for c in self.controls:
            if(c.name == ID):
                return(c)
        return(None)
